Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSubstructureSearcher.cpp

```diff
@@ -182,13 +183,25 @@ std::vector<boost::dynamic_bitset<>> getHitSynthons(
     // if the fragment didn't match anything, the whole thing's a bust.
     if (!fragMatched) {
       synthonsToUse.clear();
-      return synthonsToUse;
+      return retSynthons;
     }
   }
 
   // Fill in any synthons where they all didn't match.
   details::expandBitSet(synthonsToUse);
-  return synthonsToUse;
+  details::bitSetsToVectors(synthonsToUse, retSynthons);
+
+  // Now sort the selected synthons into ascending order of number of atoms,
+  // since smaller molecules are likely to be of more interest.
+  for (size_t i = 0; i < retSynthons.size(); ++i) {
+    std::sort(
+        retSynthons[i].begin(), retSynthons[i].end(),
+        [&](const size_t a, const size_t b) {
+          return (reaction->getSynthons()[i][a]->getOrigMol()->getNumAtoms() <
+                  reaction->getSynthons()[i][b]->getOrigMol()->getNumAtoms());
+        });
```

Comment:
I know this isn't going to be rate limiting, but this construct makes me cringe a bit.
How about (not tested):
```suggestion
    const auto &synthonsi = reaction->getSynthons()[i];
    std::sort(
        retSynthons[i].begin(), retSynthons[i].end(),
        [&](const size_t a, const size_t b) {
          return (synthonsi[a]->getOrigMol()->getNumAtoms() <
                  synthonsi[b]->getOrigMol()->getNumAtoms());
        });
```